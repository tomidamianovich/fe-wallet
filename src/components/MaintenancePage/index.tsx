import React from 'react'
import './styles/index.scss'
import { useSelector } from 'react-redux'
import { CombinedState, NavLinkState, NavLinkType } from '../../utils/type'
import { withTranslation, WithTranslation } from 'react-i18next'

type Props = {
  pageName: string
  isError?: boolean
} & WithTranslation

/*

  Component Name: MaintenancePage
  Usages: 
    - Component that allow handles maintenance and error app states

*/

const MaintenancePage: React.FC<Props> = ({ pageName, isError = false, t }) => {
  const navLinks: NavLinkState = useSelector(
    (state: CombinedState) => state?.NavLinkReducer,
  )
  const title = isError ? t('error.title') : t('maintenance.title')
  const subtitleBeginning = isError
    ? t('error.subtitle.beginning')
    : t('maintenance.subtitle.beginning')
  const subtitleEnding = isError
    ? t('error.subtitle.ending')
    : t('maintenance.subtitle.ending')
  const getPageName = () => {
    const page: NavLinkType = navLinks?.find(
      page => page?.to === pageName?.toLocaleLowerCase(),
    )
    return page?.name ?? ''
  }

  return (
    <div className="fe-wallet__maintenance-page">
      <h1>{title}</h1>
      <h3>
        {subtitleBeginning}
        {getPageName()}
        {subtitleEnding}
      </h3>
    </div>
  )
}

export default withTranslation()(MaintenancePage)
