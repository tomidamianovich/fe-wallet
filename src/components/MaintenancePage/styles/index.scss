@import '../../../containers/App/styles/colours.scss';

.fe-wallet__maintenance-page {
  display: flex;
  width: 100%;
  height: 100%;
  margin: 1rem 3rem;
  border-radius: 21px;
  padding: 4rem 0rem;
  background-color: $tertiary-base-color;
  flex-direction: column;
  align-items: stretch;
}
