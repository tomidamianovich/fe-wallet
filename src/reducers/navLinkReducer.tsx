import * as actionTypes from '../actionTypes/NavLinkType'
import { NavLinkAction, NavLinkState } from '../utils/type'
import { NAV_LINKS } from '../utils/constants'

const initialState: NavLinkState = NAV_LINKS

const reducer = (
  state: NavLinkState = initialState,
  action: NavLinkAction,
): NavLinkState => {
  switch (action.type) {
    case actionTypes.SET_NAV_LINKS:
      return action.payload
  }
  return state
}

export default reducer
