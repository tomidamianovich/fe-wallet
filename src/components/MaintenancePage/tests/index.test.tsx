import { render } from '@testing-library/react';
import MaintenancePage from '../index';
import navLinkReducer from "../../../reducers/navLinkReducer"
import { Provider } from 'react-redux';
import { createStore } from "redux";

const store = createStore(navLinkReducer);

jest.mock('react-i18next', () => ({
  withTranslation: () => (Component: any) => {
    Component.defaultProps = {...Component.defaultProps, t: (translation: string) => translation};
    return Component;
  }
}));

test('renders MaintenancePage with name', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MaintenancePage 
        pageName="wallet"
      />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders MaintenancePage with name', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MaintenancePage 
        pageName=""
      />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});