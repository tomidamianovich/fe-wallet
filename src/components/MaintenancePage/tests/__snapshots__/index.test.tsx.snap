// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MaintenancePage with name 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__maintenance-page"
  >
    <h1>
      maintenance.title
    </h1>
    <h3>
      maintenance.subtitle.beginningmaintenance.subtitle.ending
    </h3>
  </div>
</DocumentFragment>
`;

exports[`renders MaintenancePage with name 2`] = `
<DocumentFragment>
  <div
    class="fe-wallet__maintenance-page"
  >
    <h1>
      maintenance.title
    </h1>
    <h3>
      maintenance.subtitle.beginningmaintenance.subtitle.ending
    </h3>
  </div>
</DocumentFragment>
`;
